pages/v4/Coroutines.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Coroutines"
+category: "basics"
+displayOrder: 10
+version: "v4"
+---
+
+First of all, what is a Coroutine?
+
+Taken from Wikipedia:
+
+_Coroutines are very similar to threads. However, coroutines are cooperatively multitasked, whereas threads are typically preemptively multitasked. This means that coroutines provide concurrency but not parallelism. The advantages of coroutines over threads are that they may be used in a hard-realtime context (switching between coroutines need not involve any system calls or any blocking calls whatsoever), there is no need for synchronisation primitives such as mutexes, semaphores, etc. in order to guard critical sections, and there is no need for support from the operating system._
```

Comment:
Maybe shorten the quote a bit and leave out parts that aren't relevant for the use case, or too complex for beginners. Honestly, already in the second sentence where it's about "cooperatively multitasked" vs "preemptively multitaked", I'm pretty sure you'll lose a lot of people, me included 😄 Also, the hard-realtime context is just confusing, because it makes you wonder whether that is relevant to Duality, and the late mention of OS support just gets you worried about multiplatform support.